swan-lake/integration-tutorials/sending-emails-from-a-service.md

```diff
@@ -95,171 +95,127 @@ Follow the instructions given in this section to develop the service.
+        ![Define the resource](/learn/images/integration-tutorials/sending-emails-from-a-service/define_a_resource.gif)
 
-   - The initial record definitions can be generated using the "Paste JSON as record" VSCode command with the relevant JSON payloads and the records can then be modified as necessary.
+        The generated service will be as follows.
 
-6. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource function that accepts user requests, makes calls to the backend services to retrieve relevant details, and sends emails to the client with appointment details.
+        ```ballerina
+        service /healthcare on new http:Listener(8290) {
+            resource function post categories/[string category]/reserve(ReservationRequest payload) 
+                    returns http:Created|http:NotFound|http:InternalServerError {
+                
+            }
+        }
+        ```
+
+5. Define [configurable variables](https://ballerina.io/learn/by-example/#configurability) for the port on which the listener should listen and the URLs of the backend services. Also, define configurable variables for the host, username, and password of the SMTP client.
+
+    ![Define a configurable variable](/learn/images/integration-tutorials/sending-a-message-to-a-service/define_a_configurable_variable.gif)
+
+    The generated code will be as follows.
 
     ```ballerina
-    service /healthcare on new http:Listener(port) {
-        resource function post categories/[string category]/reserve(ReservationRequest payload) 
-                returns http:Created|http:NotFound|http:InternalServerError {
-            
-        }
-    }
+    configurable string hospitalServicesBackend = "http://localhost:9090";
+    configurable string paymentBackend = "http://localhost:9090/healthcare/payments";
+    configurable string host = "smtp.gmail.com";
+    configurable string username = ?;
+    configurable string password = ?;
     ```
 
-   - Use `/healthcare` as the service path (or the context) for the service attached to the listener that is listening on port `port`.
-     
-   - The HTTP resource allows the `POST` operation on resource path `/categories/{category}/reserve`, where `category` (corresponding to the specialization) is a path parameter.
+    > **Note:** Enable two factor authentication on your Google account, generate an app password, and use the app password in place of your email password. The app password can be generated with [this link](https://myaccount.google.com/apppasswords?pli=1&rapt=AEjHL4Mf5XDD4rE79YJpP5E2NoNwhvXMET_TWyBcQRn-HMzt0PI8BmptpMGRiBVIamW-0ECgVZtXxMRA19bL4Wfnq_hmjBEMqA).
+
+6. Define two [`http:Client`](https://ballerina.io/learn/by-example/#http-client) clients to send requests to the backend services and one [`email:SmtpClient`](https://ballerina.io/learn/by-example/#email-client) client to send emails.
```

Comment:
Is it a tooling issue that we have to type in the url instead of being able to pick from suggestions? Please create an issue if so.

cc @dulajdilshan

Reply:
Yeah. Tooling support needs to be improved by allowing haveing suggestions. Better to track via a GH isue